util/topsql/reporter/datasink.go

```diff
@@ -23,6 +23,6 @@ type DataSink interface {
 	// the specified deadline, or the sink is closed, an error will be returned.
 	TrySend(data *ReportData, deadline time.Time) error
 
-	// CloseDataSink notifies DataSink that the reporter is dropping it.
-	CloseDataSink()
+	// OnDeregisterFromReporter notifies DataSink that the reporter is closing and have to deregister the datasink.
+	OnDeregisterFromReporter()
```

Comment:
For this semantic, then I recommend to call it:

```suggestion
	// OnReporterClosing notifies DataSink that the reporter is closing.
	OnReporterClosing()
```